src/test/java/org/datadog/jmxfetch/TestGCMetrics.java

```diff
@@ -0,0 +1,307 @@
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_cycles_collection_time", zgcCycles);
+        }
+    }
+    
+//    @Ignore("Can not force ZGC to work using MisbehavingJMXServer")
+    @Test
+    public void testDefaultNewGCMetricsUseZGCOld() throws IOException, InterruptedException {
+        try (final MisbehavingJMXServer server = new MisbehavingJMXServer(
+                MisbehavingJMXServer.JDK_17,
+                "-XX:+UseZGC",
+                RMI_PORT,
+                CONTROL_PORT,
+                SUPERVISOR_PORT)) {
+            server.start();
+            final String ipAddress = server.getIp();
+            this.initApplicationWithYamlLines("init_config:",
+                    "  is_jmx: true",
+                    "  new_gc_metrics: true",
+                    "",
+                    "instances:",
+                    "    -   name: jmxint_container",
+                    "        host: " + ipAddress,
+                    "        collect_default_jvm_metrics: true",
+                    "        max_returned_metrics: 300000",
+                    "        port: " + RMI_PORT);
+            // Run one iteration first
+            this.app.doIteration();
+            // And then pull get the metrics or else reporter does not have correct number of metrics
+            ((ConsoleReporter) appConfig.getReporter()).getMetrics();
+
+            // Actual iteration we care about
+            this.app.doIteration();
+            final List<Map<String, Object>> actualMetrics = ((ConsoleReporter) appConfig.getReporter()).getMetrics();
+            assertThat(actualMetrics, hasSize(13));
+            final List<String> zgcPause = Collections.singletonList(
+                    "ZGC Pauses");
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_pauses_collection_count", zgcPause);
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_pauses_collection_time", zgcPause);
+            final List<String> zgcCycles = Collections.singletonList(
+                    "ZGC Cycles");
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_cycles_collection_count", zgcCycles);
+            assertGCMetric(actualMetrics, "jvm.gc.zgc_cycles_collection_time", zgcCycles);
+        }
+    }
+
+    private static void assertGCMetric(final List<Map<String, Object>> actualMetrics,
+        final String expectedMetric,
+        final List<String> gcGenerations) {
+        final List<Map<String, Object>> filteredMetrics = new ArrayList<>();
+        for (Map<String, Object> actualMetric : actualMetrics) {
+            final String name = (String) actualMetric.get("name");
+            if(expectedMetric.equals(name)) {
+                filteredMetrics.add(actualMetric);
+            }
+        }
+        assertThat(filteredMetrics, hasSize(gcGenerations.size()));
+        for (final String name : gcGenerations) {
+            log.debug("Asserting for metric '{}'", name);
```

Comment:
I wanted to not use it because it has global state. At some point I want to refactor it so that we don't use the global `metrics` state in `TestCommon`. Ideally all those functions should be static.